elfloader-tool/src/arch-riscv/crt0.S

```diff
@@ -88,8 +94,8 @@ hsm_switch_hart:
   li a7, SBI_HSM_BASE
   li a6, SBI_HSM_BASE_HART_START
   li a0, CONFIG_FIRST_HART_ID /* hart id to start */
+  mv a2, s2 /* dtb address to be passed in a1 when new hart starts is 3rd parameter */
```

Comment:
to be passed in `a1` (as the comment says) or in `a2` as the code does?

Reply:
From what I can tell, the comment is probably the one that is wrong. We seem to be calling with parameters `a0 = CONFIG_FIRST_HART_ID, a1= _start1, a2=dtb`, which would make sense.

Reply:
The third parameter of the sbi call specifies the value of a1 when the new Hart starts. So a2 holds the a1 value

Reply:
Ok, so after the call `a1` will be `dtb`. Happy with that.